### PR TITLE
Listener Registration

### PR DESCRIPTION
**Pull request by sdjmchattie**
This ensures we register for listening to messages when the web socket is opened and that we unregister again when the web socket closes.  This prevents us registering multiple times for listening to messages which was causing duplicate responses in the timesheet bot.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,77 +1,79 @@
...
+
+  util = {
+    config: require('./util/config'),
+    command: require('./util/command')
+  },
+
+  dispatcher = require('./dispatcher'),
+  rtmClient, webClient;
+
+var _handleMessage = function (message) {
```
Comment (sdjmchattie) [inline index.js]: This is the start of what I've changed.  The rest is formatting.

File: index.js [hunk 2]
```diff
@@ -1,77 +1,79 @@
...
+    logger.info("RTM client's web socket did open. Registering message listener");
+    rtmClient.addListener(RTM_EVENTS.MESSAGE, _handleMessage);
+  });
+
+  rtmClient.on(CLIENT_EVENTS.RTM.WS_CLOSE, function () {
+    logger.info("RTM client's web socket did close. Unregistering message listener.");
+    rtmClient.removeListener(RTM_EVENTS.MESSAGE, _handleMessage);
+  });
+
+  rtmClient.start();
```
Comment (sdjmchattie) [inline index.js]: This is the end of what I've changed  in this file.  The rest is formatting.

File: index.js [hunk 3]
```diff
@@ -1,77 +1,79 @@
...
+  opts.autoReconnect = opts.autoReconnect || true;
+  opts.logLevel = opts.logLevel || 'error';
+  opts.dataStore = new MemoryDataStore({});
+
+  rtmClient = new RtmClient(token, opts);
+  webClient = new WebClient(token, opts);
+
+  util.config.init(config);
+
+  _startClient();
```
Comment (sdjmchattie) [inline index.js:65]: With the exception of this line, of course.

File: dispatcher.js [hunk 4]
```diff
@@ -1,28 +1,28 @@
 'use strict';
 
 var logger = require('winston'),
-
```
Comment (sdjmchattie) [inline dispatcher.js:4]: Changes in this file are all just formatting apart from a typo in a comment.

**Review by chris-shortall — APPROVED** [review]

**Pull request merged**